server/controllers/EventController.js

```diff
@@ -17,6 +17,71 @@ const getEvents = async (req, res, next) => {
   return res.json(events)
 }
 
+/*
+* Returns an array of JSON objects of a given length (10 if undefined), filled firs by all events that still need submitting, with the rest
+* of the array being filled with the most recent events ordered by date.
+* */
+const getEventDashboardList = async (req, res, next) => {
+
+  // Check if an array length number has been defined and store that value, otherwise default the value to 10.
+  let number = (req.query.number !== undefined) ? req.query.number : 10
+
+  // Find & store all events from the collection.
+  let allEvents = await getEventsFromDb()
+
+
+
+  // Find & store all events that still need submitting.
+  let noRecordEvents = allEvents.filter((event) => {
+    return event.hasAttendanceRecords === false
+  })
+
+  // Find & store all events that have allready been submitted.
+  let recordEvents = allEvents.filter((event) => {
+    return event.hasAttendanceRecords === true
+  })
+
+  let sortedRecordEvents = recordEvents.sort((a, b) => {
+    const dateA = new Date(a.date)
+    const dateB = new Date(b.date)
+    return (dateA > dateB) ? -1 : 1
+  })
+
+  // Find how many recent events will fit into the final array, and reduce the array to only contain those objects.
+  if (noRecordEvents.length >= number) {
+    recordEvents = []
+  } else if (recordEvents.length > number - noRecordEvents.length) {
+    recordEvents = recordEvents.slice(0, number - noRecordEvents.length)
+  }
+
+  let currentEvent;
+
+  let combinedEvents = [...recordEvents, ...noRecordEvents]
+
+  if (req.query.eventId !== null) {
+    let doesExist = combinedEvents.some(event => {
+      return event._id == req.query.eventId
```

Comment:
Also can be in one line to simplify

`const doesExist = combinedEvents.some(event => event._id === req.query.eventId)`